src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -621,52 +660,73 @@ public void WriteDatabaseRecord(DatabaseRecord databaseRecord, SmugglerProgressB
                     progress.ExternalReplicationsUpdated = true;
                 }
 
-                if (currentDatabaseRecord?.RavenEtls.Count == 0 &&
-                    databaseRecord?.RavenEtls.Count > 0)
+                if (databaseRecord?.RavenEtls.Count > 0)
                 {
                     if (_log.IsInfoEnabled)
                         _log.Info("Configuring raven etls configuration from smuggler");
                     foreach (var etl in databaseRecord.RavenEtls)
                     {
+                        currentDatabaseRecord?.RavenEtls.ForEach(x =>
+                        {
+                            if (x.Name.Equals(etl.Name, StringComparison.InvariantCultureIgnoreCase))
+                            {
+                                tasks.Add(_database.ServerStore.SendToLeaderAsync(new DeleteOngoingTaskCommand(x.TaskId, OngoingTaskType.RavenEtl, _database.Name)));
+                            }
+                        });
+                        etl.TaskId = 0;
                         etl.Disabled = true;
                         tasks.Add(_database.ServerStore.SendToLeaderAsync(new AddRavenEtlCommand(etl, _database.Name)));
                     }
                     progress.RavenEtlsUpdated = true;
                 }
 
-                if (currentDatabaseRecord?.SqlEtls.Count == 0 &&
-                    databaseRecord?.SqlEtls.Count > 0)
+                if (databaseRecord?.SqlEtls.Count > 0)
                 {
                     if (_log.IsInfoEnabled)
                         _log.Info("Configuring sql etls configuration from smuggler");
                     foreach (var etl in databaseRecord.SqlEtls)
                     {
+                        currentDatabaseRecord?.SqlEtls.ForEach(x =>
+                        {
+                            if (x.Name.Equals(etl.Name, StringComparison.InvariantCultureIgnoreCase))
+                            {
+                                tasks.Add(_database.ServerStore.SendToLeaderAsync(new DeleteOngoingTaskCommand(x.TaskId, OngoingTaskType.SqlEtl, _database.Name)));
+                            }
+                        });
+                        etl.TaskId = 0;
                         etl.Disabled = true;
                         tasks.Add(_database.ServerStore.SendToLeaderAsync(new AddSqlEtlCommand(etl, _database.Name)));
                     }
                     progress.SqlEtlsUpdated = true;
                 }
 
-                if (currentDatabaseRecord?.Revisions == null &&
-                    databaseRecord?.Revisions != null)
+                if (databaseRecord?.Revisions != null)
                 {
+                    if (currentDatabaseRecord?.Revisions != null)
+                    {
+                        foreach (var collection in currentDatabaseRecord.Revisions.Collections)
+                        {
+                            if ((databaseRecord.Revisions.Collections.ContainsKey(collection.Key)) == false)
```

Comment:
Do we need to do OridinalIgnoreCase here?

Reply:
No